admin/billing/payment/stripe_webhook.go

```diff
@@ -0,0 +1,139 @@
+package payment
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/stripe/stripe-go/v79"
+	"github.com/stripe/stripe-go/v79/webhook"
+	"go.uber.org/zap"
+)
+
+// handleWebhook handles incoming webhook events from Stripe
+func (s *Stripe) handleWebhook(w http.ResponseWriter, r *http.Request) {
+	const MaxBodyBytes = int64(65536)
+	r.Body = http.MaxBytesReader(w, r.Body, MaxBodyBytes)
+	payload, err := io.ReadAll(r.Body)
+	if err != nil {
+		s.logger.Error("error reading request body", zap.Error(err))
+		http.Error(w, "error reading request body", http.StatusServiceUnavailable)
+		return
+	}
+
+	endpointSecret := s.webhookSecret
+	sigHeader := r.Header.Get("Stripe-Signature")
+	event, err := webhook.ConstructEventWithOptions(payload, sigHeader, endpointSecret, webhook.ConstructEventOptions{IgnoreAPIVersionMismatch: true})
+	if err != nil {
+		s.logger.Error("error verifying webhook signature", zap.Error(err))
+		http.Error(w, "error verifying webhook signature", http.StatusBadRequest)
+		return
+	}
+
+	s.logger.Debug(fmt.Sprintf("got event : %s\n", event.Data))
```

Comment:
Mistakenly committed? It's not clear this relates to stripe

Reply:
was looking at all events that we receive, left as it was debug level, will remove it.